consensus/consensus-types/src/vote_msg.rs

```diff
@@ -54,6 +55,19 @@ impl VoteMsg {
             self.vote().epoch() == self.sync_info.epoch(),
             "VoteMsg has different epoch"
         );
+
+        // Let us verify the vote data
+        self.vote.vote_data().verify().map_err(|e| {
```

Comment:
why not have that in Vote.verify() ?

Reply:
+ 1 to this, also I don't think we should send the SecurityEvent here, it's unified at the upper layer to be InvalidConsensusMsg

Reply:
yup no need for a log here

Reply:
Yeah, I knew this last week and had time to forget about it over the weekend.

It's nice it allows me to remove the Security event from the list and also to avoid introducing a new dependency in consensus-type. :)